packages/@lwc/template-compiler/src/parser/parser.ts

```diff
@@ -45,6 +46,12 @@ interface ParentWrapper {
     current: ParentNode;
 }
 
+interface Scope {
```

Comment:
Enhanced the parser scope to keep track of the context we need from any previous sibling scopes + any context we need to preserve about our current scope. This is currently only used for keeping track of lwc:if, lwc:elseif, and lwc:else relationships.